contracts/refs/CoreRefV2.sol

```diff
@@ -0,0 +1,241 @@
+                _core.hasRole(role5, msg.sender),
+            "UNAUTHORIZED"
+        );
+        _;
+    }
+
+    /// @notice set pausable methods to paused
+    function pause() public override onlyGuardianOrGovernor {
+        _pause();
+    }
+
+    /// @notice set pausable methods to unpaused
+    function unpause() public override onlyGuardianOrGovernor {
+        _unpause();
+    }
+
+    /// @notice address of the Core contract referenced
+    /// @return ICore implementation address
+    function core() public view override returns (ICoreV2) {
+        return _core;
+    }
+
+    /// @notice address of the Volt contract referenced by Core
+    /// @return IVolt implementation address
+    function volt() public view override returns (IVolt) {
+        return _core.volt();
+    }
+
+    /// @notice address of the Vcon contract referenced by Core
+    /// @return IERC20 implementation address
+    function vcon() public view override returns (IERC20) {
+        return _core.vcon();
+    }
+
+    /// @notice volt balance of contract
+    /// @return volt amount held
+    function voltBalance() public view override returns (uint256) {
+        return volt().balanceOf(address(this));
+    }
+
+    /// @notice vcon balance of contract
+    /// @return vcon amount held
+    function vconBalance() public view override returns (uint256) {
+        return vcon().balanceOf(address(this));
+    }
+
+    /// @notice set new reference to core
+    /// only callable by governor
+    /// @param newCore to reference
+    function setCore(address newCore) external onlyGovernor {
+        address oldCore = address(_core);
+        _core = ICoreV2(newCore);
+
+        emit CoreUpdate(oldCore, newCore);
+    }
+
+    /// @notice sweep target token, this shouldn't be needed, however it is a backup
+    /// in case a contract holds tokens and isn't a PCV Deposit
+    /// @param token to sweep
+    /// @param to recipient
+    /// @param amount of token to be sent
+    function sweep(
+        address token,
+        address to,
+        uint256 amount
+    ) external virtual onlyPCVController {
+        IERC20(token).safeTransfer(to, amount);
+    }
+
+    /// ------------------------------------------
+    /// ------------ Emergency Action ------------
+    /// ------------------------------------------
+
+    /// inspired by MakerDAO Multicall:
+    /// https://github.com/makerdao/multicall/blob/master/src/Multicall.sol
+
+    /// @notice struct to pack calldata and targets for an emergency action
+    struct Call {
+        address target;
+        uint256 value;
+        bytes callData;
+    }
+
+    /// TODO add testing here that both sends and does not send eth
+
+    /// @notice due to inflexibility of current smart contracts,
+    /// add this ability to be able to execute arbitrary calldata
+    /// against arbitrary addresses.
+    /// callable only by governor
+    function emergencyAction(Call[] calldata calls)
```

Comment:
Do you really need this arbitrary general ability to make any call to any contract that inherits `CoreRefV2`? Why not scope this and just add it to contracts where it's necessary? As with the sweep function, it's a pretty significant expansion of governance powers